### PR TITLE
#1451 - research logs

### PR DESCRIPTION
**Pull request by ElenaPoduzova**
[#1451](https://github.com/jdi-testing/jdi-light/issues/1451) - fix logging problems: 
- add loging in failure method, 
- move allure logs to separate class, 
- add allure logs to before and after handlers - so, in allure report every step has correct execution time, - move screenshot.strategy parameter to ScreenshotMaker and add method takeScreenOnFailure.

### CODE REVIEW

**Comment by codacy-review** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- jdi-light/src/main/java/com/epam/jdi/light/logger/AllureLoggerHelper.java  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/42_/jdi-light/pullRequest?prid=4646961&bid=15562338)

File: jdi-light-examples/src/test/java/io/github/epam/testng/TestNGListener.java [hunk 1]
```diff
@@ -5,13 +5,15 @@
...
 
+import com.epam.jdi.light.logger.AllureLoggerHelper;
 import org.testng.IInvokedMethod;
 import org.testng.IInvokedMethodListener;
 import org.testng.ITestResult;
 import org.testng.annotations.Test;
 
 import java.lang.reflect.Method;
 
+import static com.epam.jdi.light.driver.ScreenshotMaker.takeScreen;
```
Comment (codacy-review) [inline jdi-light-examples/src/test/java/io/github/epam/testng/TestNGListener.java]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.light.driver.ScreenshotMaker.takeScreen'](https://app.codacy.com/app/42_/jdi-light/pullRequest?prid=4646961)

**Comment by codacy-review** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- jdi-light/src/main/java/com/epam/jdi/light/logger/AllureLoggerHelper.java  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/42_/jdi-light/pullRequest?prid=4646961&bid=15562338)

File: jdi-light-examples/src/test/java/io/github/epam/testng/TestNGListener.java [hunk 2]
```diff
@@ -5,13 +5,15 @@
  * Email: roman.iovlev.jdi@gmail.com; Skype: roman.iovlev
  */
 
+import com.epam.jdi.light.logger.AllureLoggerHelper;
```
Comment (codacy-review) [inline jdi-light-examples/src/test/java/io/github/epam/testng/TestNGListener.java]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.light.logger.AllureLoggerHelper'](https://app.codacy.com/app/42_/jdi-light/pullRequest?prid=4646961)

File: jdi-light-bootstrap-tests/src/test/java/io/github/epam/testng/TestNGListener.java [hunk 3]
```diff
@@ -5,6 +5,7 @@
  * Email: roman.iovlev.jdi@gmail.com; Skype: roman.iovlev
  */
 
+import com.epam.jdi.light.logger.AllureLoggerHelper;
```
Comment (codacy-review) [inline jdi-light-bootstrap-tests/src/test/java/io/github/epam/testng/TestNGListener.java]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.light.logger.AllureLoggerHelper'](https://app.codacy.com/app/42_/jdi-light/pullRequest?prid=4646961)

**Comment by codacy-review** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- jdi-light/src/main/java/com/epam/jdi/light/logger/AllureLoggerHelper.java  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/42_/jdi-light/pullRequest?prid=4646961&bid=15562338)

File: jdi-light-examples/src/test/java/io/github/epam/testng/TestNGListener.java (hunk 1 above)
Comment (codacy-review) [inline jdi-light-examples/src/test/java/io/github/epam/testng/TestNGListener.java]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.light.driver.ScreenshotMaker.takeScreen'](https://app.codacy.com/app/42_/jdi-light/pullRequest?prid=4646961)

File: jdi-light-examples/src/test/java/io/github/epam/testng/TestNGListener.java (hunk 2 above)
Comment (codacy-review) [inline jdi-light-examples/src/test/java/io/github/epam/testng/TestNGListener.java]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.light.logger.AllureLoggerHelper'](https://app.codacy.com/app/42_/jdi-light/pullRequest?prid=4646961)

File: jdi-light-bootstrap-tests/src/test/java/io/github/epam/testng/TestNGListener.java (hunk 3 above)
Comment (codacy-review) [inline jdi-light-bootstrap-tests/src/test/java/io/github/epam/testng/TestNGListener.java]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.light.logger.AllureLoggerHelper'](https://app.codacy.com/app/42_/jdi-light/pullRequest?prid=4646961)

**Comment by codacy-review** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- jdi-light/src/main/java/com/epam/jdi/light/logger/AllureLoggerHelper.java  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/42_/jdi-light/pullRequest?prid=4646961&bid=15562338)

**Comment by mf-code** [comment]
[4010] -

**Comment by mf-code** [comment]
[4010] -

**Comment by mf-code** [comment]
[4010] -

**Comment by mf-code** [comment]
[4010] - Allure report on Netlify: https://5df1320ea8ddee01f9b35eea--musing-swartz-6fc060.netlify.com

**Comment by mf-code** [comment]
[4012] -

**Comment by mf-code** [comment]
[4012] -

**Comment by mf-code** [comment]
[4012] -

**Comment by mf-code** [comment]
[4012] - Allure report on Netlify: https://5df13b553e265101c30f8fc2--musing-swartz-6fc060.netlify.com

**Pull request merged**